docs/HYPERVISORS.md

```diff
@@ -0,0 +1,85 @@
+# EVE's use of hypervisors and hardware assisted virtualization
+
+EVE is slightly unusual in how it looks at hypervisors and hardware assisted virtualization technology. Unlike some of the other systems that were clearly built as virtualization platforms (ESXi, Hyper-V, etc.) EVE's ultimate task is to be able to run [Edge Containers](ECOS.md) as efficiently (and securely!) as possible. A running ECO is represented as a series of tasks consuming resources provided by each Edge Node (such as CPU, RAM, volumes, network instances, etc.). Tasks can further improve their performance (and security) by requesting accelerator and/or isolation services typically provided by type-1 or type-2 hypervisors and backed by hardware assisted virtualization.
+
+This mapping of Tasks to resources creates an interesting architecture where things like virtualization capabilities can be consumed transparently without the user explicitly tweaking the virtualization knobs like it would be the case with systems like libvirt/virsh.
+
+The rest of this document summarizes various implementation details we leverage for providing acceleration and isolation services to tasks.
+
+## Device models
+
+Whenever a task gets isolated into a standalone domain (either for performance or security reasons) domain gets assigned a certain, fixed amount of RAM and can have a number of virtual CPUs that would get multiplexed on top the physical CPUs available on the Edge Node. Additionally, a task may request a subset of PCI devices to be exclusively available to it (which makes them unavailable for any other domains and the host itself). Providing RAM, CPU and direct PCI assignment to a given task is one half of a job of any hypervisor. That's the easy part.
```

Comment:
> into a standalone domain (either for performance or security reasons) 

No need for braces, it takes away. Can be "into a standalone domain, for performance or security reasons, "

Reply:
> subset of PCI devices to be exclusively available to it (which makes them unavailable for any other domains and the host itself)

Sentence can be simplified as: "subset of PCI devices to be available to it exclusively, which makes them unavailable for any other domains and the host itself."

Reply:
fixed